src/main/java/ch/njol/skript/expressions/ExprHatchingType.java

```diff
@@ -0,0 +1,109 @@
+/**
+ *   This file is part of Skript.
+ *
+ *  Skript is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation, either version 3 of the License, or
+ *  (at your option) any later version.
+ *
+ *  Skript is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with Skript.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * Copyright Peter Güttinger, SkriptLang team and contributors
+ */
+package ch.njol.skript.expressions;
+
+import ch.njol.skript.Skript;
+import ch.njol.skript.bukkitutil.EntityUtils;
+import ch.njol.skript.classes.Changer.ChangeMode;
+import ch.njol.skript.doc.Description;
+import ch.njol.skript.doc.Events;
+import ch.njol.skript.doc.Examples;
+import ch.njol.skript.doc.Name;
+import ch.njol.skript.doc.Since;
+import ch.njol.skript.entity.EntityData;
+import ch.njol.skript.lang.Expression;
+import ch.njol.skript.lang.ExpressionType;
+import ch.njol.skript.lang.SkriptParser.ParseResult;
+import ch.njol.skript.lang.util.SimpleExpression;
+import ch.njol.util.Kleenean;
+import ch.njol.util.coll.CollectionUtils;
+import org.bukkit.entity.EntityType;
+import org.bukkit.event.Event;
+import org.bukkit.event.player.PlayerEggThrowEvent;
+import org.eclipse.jdt.annotation.Nullable;
+
+@Name("Hatching Entity Type")
+@Description("The type of the entity that will be hatched in a Player Egg Throw event.")
+@Examples({
+	"on player egg throw:",
+	"\tset the hatching entity type to a primed tnt"
+})
+@Events("Egg Throw")
+@Since("INSERT VERSION")
+@SuppressWarnings("rawtypes")
```

Comment:
Return EntityData<?> So we don't get the rawtypes warning.

Reply:
We can do that, but I'll have to do some casting in places that use EntityData.class due to the generics - would that be preferred?

Reply:
It's already casting to EntityData<?> at the one place just without the specified generic, which is why there is a warning because it thinks you're forgetting about it.